src/Dscp/Crypto/Impl.hs

```diff
@@ -64,3 +115,13 @@ verify = abstractVerify
 
 unsafeVerify :: HasSignature a => PublicKey -> a -> Signature b -> Bool
 unsafeVerify = unsafeAbstractVerify
+
+data Signed msg = Signed
+    { sgMessage   :: msg
+    , sgPublicKey :: PublicKey
+    , sgSignature :: Signature msg
+    } deriving (Eq, Show, Generic)
```

Comment:
Are you sure that we'd want to include a pubkey with a signature in all possible usecases?

Reply:
Yes, it should be alright.